langchain4j-core/src/test/java/dev/langchain4j/data/document/MetadataTest.java

```diff
@@ -0,0 +1,95 @@
+package dev.langchain4j.data.document;
+
+import org.assertj.core.api.WithAssertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+class MetadataTest implements WithAssertions {
+    @Test
+    public void test_shared_map_constructor() {
```

Comment:
@crutcher could you please remove this test? it is obsolete now as I have made constructor to copy the provided map

Reply:
Done